lambdas/totp-generator/src/totp-generator-handler.ts

```diff
@@ -0,0 +1,38 @@
+import { APIGatewayProxyResult } from "aws-lambda";
+import { LambdaInterface } from "@aws-lambda-powertools/commons";
+
+import TOTP from "totp-generator";
+import { Logger } from "@aws-lambda-powertools/logger";
+const TOTP_TTL_IN_SECONDS = 30;
+const TOTP_HASH = "SHA-512";
+const TOTP_LENGTH = 8;
+
+const logger = new Logger();
+
+export class TotpGeneratorHandler implements LambdaInterface {
+  public async handler(
+    event: any,
+    _context: unknown
+  ): Promise<APIGatewayProxyResult | any> {
+    try {
+      if (!event.SecretString) {
```

Comment:
I suppose for now this is Ok, but when this is part of step-function we wouldn't need this since it's output from the integration step above it i.e getSecretValue integration always has an output  `event.SecretString` when things are fine when a failure happens it would go into it step function catch flow

Reply:
There's a chance it could be an empty string though right? If the parameter initialisation doesn't work - the secret is created but it's blank

Reply:
Then the function wouldn't be called at all, in stepfunction b'cos the getSecretValue would fail, it should be in it catch

Reply:
I'm basically just saying we should keep it aha